location/models.py

```diff
@@ -160,12 +160,33 @@ def get_location_from_ids(self, qsr, loc_type):
         return list(qsr)
     
     def is_allowed(self, user, locations_id):
+        if user.is_superuser or not settings.ROW_SECURITY:
+            return True 
         if hasattr(user, '_u'):
             user = user._u
         cache_name = f'user_locations_{user.id}'
         allowed = cache.get(cache_name)
         if not allowed:
-            allowed = list(self.allowed(user.id, qs=None))
+            # for CA
+            if user.is_claim_admin and user.health_facility:
+                allowed = list(self.children(
+                    user.id, 
+                    location_id=user.health_facility.location_id
+                ))
+            if user.is_officer:
```

Comment:
@delcroip can a claim admin also be an officer? If not consider making this `elif` as `allowed` is overridden anyway when both `if` conditions are satisfied.

Reply:
in Legacy no, in Modular it is technically possible but when writing the code I realized that  current  implementation of the user location need a full rework. I think we should Drop all together OfficerVillage and ClaimAdmin and Officer table, I need to write proposed specs for that

long story short I see something like:

- General location Auth based on UserDisctrict renamed UserLocation, this could have V, W, D, R location type
- Hybrid authority: a new table call PolicyHolderUser  renamed to UserCompositeRightsDetails 
    -  generic Fk (to location, PH ... )
    -  user (fk to user)
    -  compositeType (string)